### PR TITLE
Add automatic axon segmentation training

### LINKED ISSUE #3: Try fully automatic axon segmentation
We would like to know how easily SAM can be fine-tuned for fully automatic segmentation (i.e. prompting with the whole image as a bbox instead of prompting with a ROI of interest).

The perfect pretext to try this is axon segmentation.
1. If it works well, this axon segmentation could then be used to generate the bbox prompts used subsequently for the myelin segmentation
2. There is no overlap between instances in the axon class. The myelin class would not be well suited because there are many overlaps (or rather touching myelin sheaths). A big advantage of segmenting the myelin with localized bbox prompts is that we get a clean and reliable instance segmentation. The way AxonDeepSeg currently works it that the semantic segmentation is "subdivided" (semantic to instance) with a watershed algorithm. Although it works fine in a lot of cases, sometimes this process deteriorates the segmentation. See example below, and look for small axons touching big axons. In the instance segmentation, a "leak" artifact occurs, where the myelin of the small axon is wrongly attributed to the big axon.
image

### PR DESCRIPTION
**Pull request by hermancollin**
Fully automatic segmentation is performed by prompting SAM with the full image bounding box. Currently this is what we get after 100 epochs (**validation** image).

sub-nyuMouse10_sample-0008_TEM_val-seg-axon_epoch95
losses_axon_seg

In the SAM paper, fully automatic segmentation is performed by prompting with a grid of points. Maybe this would be worth trying because that's how it was originally designed.

Resolves #3

### CODE REVIEW

**Comment by hermancollin** [comment]
Better results with minor hyperparameters optimization in c1a69307bb2ece69eee4bf1d02516d194d2a4520, namely more agressive LR, more epochs
sub-nyuMouse10_sample-0008_TEM_val-seg-axon_epoch145

losses_axon_seg_150epoch_lr1e-4

**Comment by hermancollin** [comment]
~~I also tried using a bigger model (ViT_H instead of ViT_B) but it doesn't seem to help (on the contrary, it almost looks worse now; looks like overfit because the training loss actually went down while the validation seg quality decreased)~~
sub-nyuMouse10_sample-0008_TEM_val-seg-axon_epoch145
losses_axon_seg_vit_h


Here is the GT for reference:
sub-nyuMouse10_sample-0008_TEM_seg-axonmyelin-manual

**EDIT** actually this was not done properly; see #6
